### PR TITLE
Add %specpartsdir to macros.in

### PR DESCRIPTION
**Pull request by ffesti**
so it can be configured from the spec file

Also move the dir beside the %buildsubdir as %{buildsubdir}-SPECPARTS so it doesn't pollute the %buildsubdir.

This patch is still missing adjustments to the dynamic_specs.md as we might decide to keep the SPECPARTS in %buildsubdir

### CODE REVIEW

**Comment by ffesti** [comment]
Replaces: #2533
Resolves: #2532

**Comment by pmatilai** [comment]
Right, that's another way to deal with it. Having an extra directory seems a bit messy but then it'd be just temporary thing until (and further motivation for) #2078 . 

It does add an unconditional dependency on using a buildsubdir (and so, %setup), but that's not necessarily a bad thing at all.

File: macros.in [hunk 1]
```diff
@@ -132,6 +132,8 @@
 
 %_keyringpath		%{_dbpath}/pubkeys/
 
+%specpartsdir %{_builddir}/%{buildsubdir}-SPECPARTS
```
Comment (pmatilai) [inline macros.in:135]: Move this next to the other build directory stuff, _specdir and buildroot and whatnot.

**Comment by pmatilai** [comment]
It's of course not actually introduced in this patch, but I have to say the `%specpartsdir` macro name continues to look odd without a leading underscore. But probably that's just  from years of conditioning from exposure to the weird underscorey names :laughing:

File: macros.in (hunk 1 above)
Comment (hroncok) [reply macros.in:135]: Is buildsubdir always defined?

Comment (pmatilai) [reply macros.in:135]: In practise yes, because the debuginfo machinery depends on it.

Comment (pmatilai) [reply macros.in:135]: Oh and just realized that moving the specparts dir out of buildsubdir makes it kinda unnecessary to make it user overridable, but maybe there are other use-cases for that anyhow.

**Comment by ffesti** [comment]
Yes, the underscores in macros are a bit of a mess. I modeled %specpartsdir on %buildsubdir which also comes without an underscore. I think the idea is that macros to be used directly don't have an underscore. Stuff to be configured have one underscore and stuff not to be touched have two.

**Pull request merged**